File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Model/Entities/DocumentUnits/DocumentUnitModel.cs
using System;
using System.Collections.Generic;
using VecompSoftware.Commons.Interfaces.CQRS.Commands;
using VecompSoftware.DocSuiteWeb.Model.Entities.Commons;
using VecompSoftware.DocSuiteWeb.Model.Entities.Fascicles;
using VecompSoftware.DocSuiteWeb.Model.Entities.Tenants;
using VecompSoftware.DocSuiteWeb.Model.Parameters;

namespace VecompSoftware.DocSuiteWeb.Model.Entities.DocumentUnits
{
    public class DocumentUnitModel : IContentBase
    {
        #region [ Constructor ]
        public DocumentUnitModel()
        {
            UniqueId = Guid.NewGuid();
        }
        #endregion

        #region [ Fields ]

        #endregion

        #region [ Properties ]

        public Guid UniqueId { get; set; }

        public int? EntityId { get; set; }

        public string DocumentUnitName { get; set; }

        public short Year { get; set; }

        public string Number { get; set; }

        public string Title { get; set; }

        public ReferenceType? ReferenceType { get; set; }

        public DateTimeOffset? RegistrationDate { get; set; }

        public string RegistrationUser { get; set; }

        public string Subject { get; set; }

        public Guid? IdFascicle { get; set; }

        public bool? IsFascicolable { get; set; }

        public CategoryModel Category { get; set; }

        public ContainerModel Container { get; set; }

        public int Environment { get; set; }

        public string MainDocumentName { get; set; }

        public Guid? IdUDSRepository { get; set; }

        public TenantAOOModel TenantAOO { get; set; }

        public DocumentUnitChainModel DocumentUnitChain { get; set; }

        public ICollection<DocumentUnitRoleModel> DocumentUnitRoles { get; set; }

        public FascicleFolderModel FascicleFolder { get; set; }

        #endregion
    }
}

File: SOURCE_DOCSUITE/VecompSoftware.DocSuiteWeb.Gui/App/Models/DocumentUnits/DocumentUnitModel.ts
import CategoryModel = require('App/Models/Commons/CategoryModel');
import ContainerModel = require('App/Models/Commons/ContainerModel');
import FascicleReferenceType = require('App/Models/Fascicles/FascicleReferenceType');
import UDSRepositoryModel = require('App/Models/UDS/UDSRepositoryModel')
import DocumentUnitRoleModel = require('App/Models/DocumentUnits/DocumentUnitRoleModel');
import DocumentUnitChainModel = require('App/Models/DocumentUnits/DocumentUnitChainModel');
import RoleModel = require('../Commons/RoleModel');

class DocumentUnitModel {
    Environment: number;
    UniqueId: string;
    EntityId: number;
    DocumentUnitName: string;
    Year: number;
    Number: string;
    Title: string;
    ReferenceType: FascicleReferenceType;
    RegistrationDate: Date;
    RegistrationUser: string;
    Subject: string;
    Category: CategoryModel;
    Container: ContainerModel;
    UDSRepository: UDSRepositoryModel;
    Roles: RoleModel[];
    DocumentUnitRoles: DocumentUnitRoleModel[] = new Array<DocumentUnitRoleModel>();
    DocumentUnitChains: DocumentUnitChainModel[] = new Array<DocumentUnitChainModel>();
    IdFascicle: string;
    IsFascicolable?: boolean;

    constructor() {     
    }
}

export = DocumentUnitModel;